sspl_test/coverage/coverage_setup.py

```diff
@@ -0,0 +1,185 @@
+        include="/opt/seagate/*",
+        omit=['*/.local/*', '*/usr/*'],
+        # config_file='%scoverage/.coveragerc',
+        )\
+"""%(DATA_PATH, DATA_PATH)
+
+PATCH_2 = """\
+        logger.info("Starting coverage report scope")
+        co.start()\
+"""
+
+PATCH_3 = """\
+def generate_cov_report(signal_number, frame):
+    logger.info('Ending Coverage Scope')
+    co.stop()
+    logger.info('coverage object stopped.')
+    co.save()
+    logger.info('coverage info saved.')
+    cov_per = co.xml_report(ignore_errors=True,
+        outfile='%scoverage/sspl_xml_coverage_report.xml')
+    logger.info(f'XML coverage report generated with coverage of {cov_per} percentage.')
+    ## Enable below code to inable HTML report generation
+ 	# html_cov_per = co.html_report(
+ 	#                     directory='%scoverage/sspl_html_coverage',
+ 	#                     ignore_errors=True,
+ 	#                 )
+ 	# logger.info(f'HTML coverage report geverated with coverage of {html_cov_per} percentage.')\
+"""%(DATA_PATH, DATA_PATH)
+
+PATCH_4 = """\
+    signal.signal(signal.SIGUSR1, generate_cov_report)\
+"""
+
+def coverage_setup():
+    """Installs pip3 coverage package. Injects different patches code from
+       coverage_code file to sspl_ll_d file, also creates target directory
+       for code coverage report and assigns permission to the directory.
+    """
+    print("Installing coverage.py")
+    cmd = 'python3 -m pip install coverage'
+    _, err, return_code = SimpleProcess(cmd).run()
+    if return_code:
+        print(err)
+        return return_code
+
+    print("Creating required files for coverage..")
+    patch1_name = "initialize coverage obj for code coverage report generation"
+    patch2_name = "start the code coverage scope"
+    patch3_name = "stop coverage, save and generate code coverage report"
+    patch4_name = "signal handler for SIGUSR1 to generate code coverage report"
+
+    with open(SSPL_LL_D, 'r') as sspl_ll_d:
+        sspl_ll_d_lines = sspl_ll_d.readlines()
+
+    for i, line in enumerate(sspl_ll_d_lines):
+        if patch1_name in line:
+            for j,l in enumerate(PATCH_1.split('\n')):
+                sspl_ll_d_lines.insert(i+j+1, l+'\n')
+        elif patch2_name in line:
+            for j,l in enumerate(PATCH_2.split('\n')):
+                sspl_ll_d_lines.insert(i+j+1, l+'\n')
+        elif patch3_name in line:
+            for j,l in enumerate(PATCH_3.split('\n')):
+                sspl_ll_d_lines.insert(i+j+1, l+'\n')
+        elif patch4_name in line:
+            for j,l in enumerate(PATCH_4.split('\n')):
+                sspl_ll_d_lines.insert(i+j+1, l+'\n')
+
+    shutil.move(SSPL_LL_D, f'{SSPL_LL_D}.bak')
+
+    with open(SSPL_LL_D, 'x') as sspl_ll_d:
+        for line in sspl_ll_d_lines:
+            sspl_ll_d.write(line)
+
+    print("Adding permission to files %s, %s"%(SSPL_LL_D, REPORT_PATH))
+    uid =  pwd.getpwnam("sspl-ll").pw_uid
+    os.chmod(SSPL_LL_D, 0o755)
```

Comment:
New temp sspl_ll_d file is created, which needs to be given 755 permission, statement can't be avoided.